src/Indice.Features.Cases.App/src/app/app.links.ts

```diff
@@ -1,40 +1,98 @@
+import { CasesApiService } from './core/services/cases-api.service';
+import { Params } from '@angular/router';
 
 export class AppLinks implements IAppLinks {
 
-    private _main: ReplaySubject<NavLink[]> = new ReplaySubject<NavLink[]>(1);
-
-    constructor(private authService: AuthService) {
-        this.authService.user$.pipe(
-            map(user => {
-                const headerMenu = [
-                    new NavLink('Αρχική', '/dashboard', true),
-                    new NavLink('Υποθέσεις', '/cases', true)
-                ]
-                if (this.authService.isAdmin()) {
-                    headerMenu.push(new NavLink('Διαχείριση Υποθέσεων', '/case-types', true));
-                } else {
-                    headerMenu.push(new NavLink('Ειδοποιήσεις', '/notifications', true));
-                }
-                return headerMenu;
-            }),
-            tap(navLinks => this._main.next(navLinks))
-        ).subscribe();
-    }
-
-    public public: Observable<NavLink[]> = of([]);
-    public profileActions: Observable<NavLink[]> = of([]);
-
-    public main: Observable<NavLink[]> = this._main.asObservable();
-
-    public profile: Observable<NavLink[]> = of([
-        new NavLink('Αποσύνδεση', '/logout', false)
-    ]);
-
-    public legal: Observable<NavLink[]> = of([]);
-    public brand: Observable<NavLink[]> = of([]);
+  private _main: ReplaySubject<NavLink[]> = new ReplaySubject<NavLink[]>(1);
+
+  // constructor(private authService: AuthService, private _api: CasesApiService) {
+  //   zip(this.authService.user$.pipe(
+  //     map(user => {
+  //       const headerMenu = [
+  //         new NavLink('Αρχική', '/dashboard', true),
+  //         new NavLink('Υποθέσεις', '/cases', true)
+  //       ];
+
+  //       if (this.authService.isAdmin()) {
+  //         headerMenu.push(new NavLink('Διαχείριση Υποθέσεων', '/case-types', true));
+  //       } else {
+  //         headerMenu.push(new NavLink('Ειδοποιήσεις', '/notifications', true));
+  //       }
+
+  //       return headerMenu;
+  //     }),
+  //     tap(navLinks => this._main.next(navLinks))
+  //   ), this._api.getMenuItems().pipe(
+  //     map(data => {
+  //       return data.items?.map(item => new NavLink(`${item.title}`, `/menu-items/${item.id}`, true));
+  //     })
+  //   )
+  //   ).pipe(map((navLinks) => {
+  //     if (navLinks[1]) {
+  //       return navLinks[0].concat(navLinks[1]);
+  //     }
+  //     return navLinks[0];
+  //   })).subscribe();
+  // }
+
+  constructor(
+    private authService: AuthService,
+    private _api: CasesApiService) {
+    this.authService.user$.pipe(
+      map(user => {
+        const headerMenu = [
+          new NavLink('Αρχική', '/dashboard', true),
+          new NavLink('Υποθέσεις', '/cases', true)
+        ];
+
+        if (this.authService.isAdmin()) {
+          headerMenu.push(new NavLink('Διαχείριση Υποθέσεων', '/case-types', true));
+          this.appendMenuItems(headerMenu);
```

Comment:
@dkarkanas αν βάλει σε άλλο σημείο αυτό τον κώδικα προσθήκης των menuItems, που προκαλεί ένα authenticated call πισω στο ΑΠΙ, τότε το ΑΠΙ κωλλ γινεται χωρίς authentication αν κατάλαβα καλά απο την περιγραφή του Κωνσταντίνου. Αυτό μάλλον μας οδηγεί να διαγνώσουμε οτι αυτο το this.authService.user$ που εντός του γίνεται η κλήση προκαλή την αρχικοποίηση του authentication header κλπ στην εφαρμογή. Θα πρέπει να γίνει κάπου νωρίτερα αυτή η κλήση.